src/scrapers/visa-cal.ts

```diff
@@ -45,6 +47,94 @@ interface ScrapedAdditionalInfo {
   category?: string;
 }
 
+interface InitResponse {
+  result: {
+    cards: {
+      cardUniqueId: string;
+      last4Digits: string;
+      [key: string]: unknown;
+    }[];
+  };
+}
+type CurrencySymbol = '₪' | string;
+interface CardTransactionDetailsError {
+  title: string;
+  statusCode: number;
+}
+interface CardTransactionDetails extends CardTransactionDetailsError {
+  result: {
+    bankAccounts: {
+      bankAccountNum: string;
+      bankName: string;
+      choiceExternalTransactions: any;
+      currentBankAccountInd: boolean;
+      debitDates: {
+        basketAmountComment: unknown;
+        choiceHHKDebit: number;
+        date: string;
+        debitReason: unknown;
+        fixDebitAmount: number;
+        fromPurchaseDate: string;
+        isChoiceRepaiment: boolean;
+        toPurchaseDate: string;
+        totalBasketAmount: number;
+        totalDebits: {
+          currencySymbol: CurrencySymbol;
+          amount: number;
+        }[];
+        transactions: {
+          amtBeforeConvAndIndex: number;
+          branchCodeDesc: string;
+          cashAccManagerName: null;
+          cashAccountManager: null;
+          cashAccountTrnAmt: number;
+          chargeExternalToCardComment: string;
+          comments: [];
+          curPaymentNum: number;
+          debCrdCurrencySymbol: CurrencySymbol;
+          debCrdDate: string;
+          debitSpreadInd: boolean;
+          discountAmount: unknown;
+          discountReason: unknown;
+          immediateComments: [];
+          isImmediateCommentInd: boolean;
+          isImmediateHHKInd: boolean;
+          isMargarita: boolean;
+          isSpreadPaymenstAbroad: boolean;
+          linkedComments: [];
+          merchantAddress: string;
+          merchantName: string;
+          merchantPhoneNo: string;
+          numOfPayments: number;
+          onGoingTransactionsComment: string;
+          refundInd: boolean;
+          roundingAmount: unknown;
+          roundingReason: unknown;
+          tokenInd: 0;
+          tokenNumberPart4: '';
+          transCardPresentInd: boolean;
+          transTypeCommentDetails: [];
+          trnAmt: number;
+          trnCurrencySymbol: CurrencySymbol;
+          trnExacWay: number;
+          trnIntId: string;
+          trnNumaretor: number;
+          trnPurchaseDate: string;
+          trnType: string;
+          trnTypeCode: string;
+          walletProviderCode: 0;
+          walletProviderDesc: '';
+        }[];
+      }[];
+      immidiateDebits: { totalDebits: [], debitDays: [] };
+    }[];
+    blockedCardInd: boolean;
+  };
+  statusCode: 1;
+  statusDescription: string;
+  statusTitle: string;
+}
```

Comment:
@erikash I will appreciate your help here.
I managed to fetch the transactions per month, and now I need to convert their object to our `Transaction` object.
I hate this phase of understanding all their fields and validating it with the content I see on the website.

Can you help me with that?

Thanks!!

Reply:
Hi @baruchiro 
Sorry but I only just saw your message - sure will do this weekend

Reply:
@baruchiro I've converted the transactions here:
https://github.com/eshaham/israeli-bank-scrapers/pull/769

Tested it a little with installments, regular transactions and multiple accounts.